src/tests/multihost/ad/test_dyndns.py

```diff
@@ -0,0 +1,312 @@
+        cmd_lookup = multihost.client[0].run_command(f"nslookup {hostname}", raiseonerr=False)
+        assert cmd_lookup.returncode == 0
+        cmd_lookup_ptr = multihost.client[0].run_command(f"nslookup {ip}", raiseonerr=False)
+        assert cmd_lookup_ptr.returncode == 0
+
+    def test_0002_verify_when_dyndns_update_set_to_false(self, multihost, adjoin):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: verify when dyndns update set to false
+        :id: 0cdbd0fe-4c67-4192-b44b-04e4227b5858
+        :steps:
+          1. Join client to AD domain
+          2. Perform DNS lookup for hostname
+          3. Perform DNS lookup for IP
+        :expectedresults:
+          1. Client joins domain
+          2. Hostname resolves incorrectly
+          3. IP resolves incorrectly
+        """
+        adjoin(membersw='adcli')
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        domain = multihost.ad[0].domainname
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+        ip = str(multihost.client[0].ip)
+
+        domain_section = 'domain/{}'.format(domain)
+        sssd_params = {'dyndns_update': 'false'}
+        client.sssd_conf(domain_section, sssd_params)
+        client.clear_sssd_cache()
+
+        cmd_forward_lookup = multihost.client[0].run_command([f"nslookup {hostname}"], raiseonerr=False)
+        assert cmd_forward_lookup.returncode != 0
+        cmd_ptr_lookup = multihost.client[0].run_command([f"nslookup {ip}"], raiseonerr=False)
+        assert cmd_ptr_lookup.returncode != 0
+
+    def test_0003_verify_with_dyndns_ttl_functionality(self, multihost, adjoin, getptr):
+        """
+        :title: IDM-SSSD-TC: ad_provider: dyndns: Verify with dyndns ttl functionality
+        :id: ffb8ee9b-2a82-4917-82b0-d87fd066a24c
+        :steps:
+          1. Join client to AD domain
+          2. Configure sssd.conf with dyndns_ttl = 9200
+          3. Perform DNS lookup for hostname
+          4. Perform DNS lookup for IP
+        :expectedresults:
+          1. Client joins domain
+          2. Should succeed
+          3. Hostname resolves with 9200 ttl
+          4. IP resolves with 9200 ttl
+        """
+        adjoin(membersw='adcli')
+        ttl = '9600'
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        domain = multihost.ad[0].domainname
+        hostname_cmd = multihost.client[0].run_command('hostname', raiseonerr=False)
+        hostname = hostname_cmd.stdout_text.rstrip()
+        (ip, network, ptr) = getptr
+        cmd_change_hostname = 'hostname ' + hostname
+        multihost.client[0].run_command(cmd_change_hostname)
+        domain_section = 'domain/{}'.format(domain)
+        sssd_params = {'dyndns_ttl': ttl}
+        client.sssd_conf(domain_section, sssd_params)
+        client.clear_sssd_cache()
+
+        time.sleep(30)
```

Comment:
Do we have a util or lib function somewhere that could run the nslookup in a loop until it passes or runs too many times?  Just an idea for a way to avoid sleeps.

Reply:
Let me do that. Are you okay with approving these changes and I can make the changes on the next patch 
https://github.com/SSSD/sssd/pull/6348 ?

Reply:
I think we have to wait until @justin-stephenson does a review also.  As far as adding that to the other PR #6348, that one is currently marked closed.  And from what I can tell, it looks like it contains a lot of the same content as this PR.  Is it supposed to be a PR to handle updates after this one?

Reply:
So with the new fixes, I think we just need @justin-stephenson to approve of the ttl check added.  I'll approve from my side.

Reply:
Nice catch, I ported this test over from bash and missed it. I updated the code with -debug, providing ttl information in the output. The default ttl is 24 hours that's 86400, with the new assertions, it will check that the TTL is less than the provided within a certain range.